### PR TITLE
Added functionality to combine 2 MODS fields when present in solr metadata display

### PR DESCRIPTION
**Pull request by BrandonMorr**

### CODE REVIEW

File: template.php [hunk 1]
```diff
@@ -133,6 +133,29 @@ function chs_theme_preprocess_islandora_solr_metadata_display(&$variables) {
 }
 
 /**
+ * Implements hook_preprocess().
+ */
+function chs_theme_process_islandora_solr_metadata_display(&$variables) {
+  $solr_fields = $variables['solr_fields'];
+
+  if (isset($solr_fields['mods_accessCondition_useAndReproduction_xlinkhref_ms']) && isset($solr_fields['mods_accessCondition_useAndReproduction_ms'])) {
```
Comment (jordandukart) [inline template.php]: I wouldn't expose having the `mods_accessCondition_useAndReproduction_xlinkhref_ms` field being present in the metadata display as it clogs up the UI when it doesn't have to. Seeing as you are grabbing the value and unsetting it later you could add it to `$variables['solr_fields']` array in https://github.com/discoverygarden/chs_theme/blob/7.x/template.php#L128-Lundefined and unsetting it here again after using it.

File: template.php [hunk 2]
```diff
@@ -133,6 +133,29 @@ function chs_theme_preprocess_islandora_solr_metadata_display(&$variables) {
...
+    // Take the xlink and node values and create an anchor out of them.
+    $link = reset($solr_fields['mods_accessCondition_useAndReproduction_xlinkhref_ms']['value']);
+    $value = reset($solr_fields['mods_accessCondition_useAndReproduction_ms']['value']);
+
+    $anchor = "<a href='$link'>$value</a>";
+
+    if (!empty($anchor)) {
+      $variables['solr_fields']['mods_accessCondition_useAndReproduction_ms']['value'] = array(
```
Comment (jordandukart) [inline template.php]: This straight up stomps on any and all modification to the values that were here previously as at this point the markup is rendered. For example if you were to put truncation on this field this completely stomps it away. Hyperlinking and truncation completely cannot work in Solr metadata though so maybe this is a moot point? This still needs to be communicated to the client. Similarly if this field had multiple values (it is a ms field after all) you squashed all the values.

File: template.php [hunk 3]
```diff
@@ -133,6 +133,29 @@ function chs_theme_preprocess_islandora_solr_metadata_display(&$variables) {
...
+ */
+function chs_theme_process_islandora_solr_metadata_display(&$variables) {
+  $solr_fields = $variables['solr_fields'];
+
+  if (isset($solr_fields['mods_accessCondition_useAndReproduction_xlinkhref_ms']) && isset($solr_fields['mods_accessCondition_useAndReproduction_ms'])) {
+    // Take the xlink and node values and create an anchor out of them.
+    $link = reset($solr_fields['mods_accessCondition_useAndReproduction_xlinkhref_ms']['value']);
+    $value = reset($solr_fields['mods_accessCondition_useAndReproduction_ms']['value']);
+
+    $anchor = "<a href='$link'>$value</a>";
```
Comment (jordandukart) [inline template.php]: Any particular reason you aren't using [l](https://api.drupal.org/api/drupal/includes%21common.inc/function/l/7.x)?

**Review by jordandukart — CHANGES_REQUESTED** [review]

**Comment by BrandonMorr** [comment]
Updated pull, client confirmed that this code will suit their use case.

**Pull request merged**